core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/PartialSuccessHandler.java

```diff
@@ -0,0 +1,192 @@
+package com.microsoft.applicationinsights.internal.channel.common;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.HttpStatus;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.common.base.Optional;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+/**
+ * This class implements the retry logic for partially accepted transmissions.
+ * HTTP status code 206.
+ * <p>
+ * 
+ * @see <a href=
+ *      "https://github.com/Microsoft/ApplicationInsights-dotnet/blob/master/docs/ServerTelemetryChannel%20error%20handling.md#partialsuccesstransmissionpolicy">PartialSuccessTransmissionPolicy</a>
+ * @author jamdavi
+ *
+ */
+public class PartialSuccessHandler implements TransmissionHandler {
+
+	/**
+	 * Ctor
+	 * 
+	 * Constructs the PartialSuccessHandler object.
+	 * 
+	 * @param policy
+	 *            The {@link TransmissionPolicyManager} object that is needed to
+	 *            control the back off policy.
+	 */
+	public PartialSuccessHandler(TransmissionPolicyManager policy) {
+	}
+
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+	}
+
+	/**
+	 * Provides the core logic for the retransmission
+	 * 
+	 * @param args
+	 *            The {@link TransmissionHandlerArgs} for this transmission.
+	 * @return Returns a pass/fail for handling this transmission.
+	 */
+	boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getTransmission() != null && args.getTransmissionDispatcher() != null) {
+			switch (args.getResponseCode()) {
+			case HttpStatus.SC_PARTIAL_CONTENT:
+				BackendResponse backendResponse = getBackendResponse(args.getResponseBody());
+				List<String> originalItems = generateOriginalItems(args);
+
+				// Somehow the amount of items received and the items sent do not match
+				if (backendResponse != null && (originalItems.size() != backendResponse.itemsReceived)) {
+					InternalLogger.INSTANCE.trace(
+							"Skipping partial content handler due to itemsReceived being larger than the items sent.");
+					return false;
+				}
+
+				if (backendResponse != null && (backendResponse.itemsAccepted < backendResponse.itemsReceived)) {
+					List<String> newTransmission = new ArrayList<String>();
+					for (BackendResponse.Error e : backendResponse.errors) {
+						switch (e.statusCode) {
+						case HttpStatus.SC_REQUEST_TIMEOUT:
+						case HttpStatus.SC_INTERNAL_SERVER_ERROR:
+						case HttpStatus.SC_SERVICE_UNAVAILABLE:
+						case 429:
```

Comment:
For the 429 and 439 case, can you append those lines with comments with a "friendly name" (similar to the constants in the previous cases)

Reply:
Fixed with constants mentioned in other change.